litmus-portal/authentication/pkg/user/repository.go

```diff
@@ -0,0 +1,140 @@
+package user
+
+import (
+	"context"
+	"litmus/litmus-portal/authentication/pkg/entities"
+	"litmus/litmus-portal/authentication/pkg/utils"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"golang.org/x/crypto/bcrypt"
+)
+
+//Repository holds the mongo database implementation of the Service
+type Repository interface {
+	FindUser(user *entities.User) (*entities.User, error)
+	UpdatePassword(userPassword *entities.UserPassword, isAdminBeingReset bool) error
+	CreateUser(user *entities.User) (*entities.User, error)
+	UpdateUser(user *entities.User) (*entities.User, error)
+	IsAdministrator(user *entities.User) error
+	GetUsers() (*[]entities.User, error)
+}
+
+type repository struct {
+	Collection *mongo.Collection
+}
+
+//FindUser helps to authenticate the user
+func (r repository) FindUser(user *entities.User) (*entities.User, error) {
+	var result = entities.User{}
+	findOneErr := r.Collection.FindOne(context.TODO(), bson.M{
+		"username": user.UserName,
+	}).Decode(&result)
```

Comment:
Have you formatted using `gofmt` ?

Reply:
Yes, the whole project has been linted using `gofmt`